pkg/filesystem/subvolume.go

```diff
@@ -175,8 +177,14 @@ func listCephFSSubvolumes(ctx context.Context, clientsets *k8sutil.Clientsets, o
 			}
 			// append the subvolume which doesn't have any snapshot attached to it.
 			for _, sv := range subvol {
-				state := getSubvolumeState(ctx, clientsets, operatorNamespace, clusterNamespace, fs.Name, sv.Name, svg.Name)
-
+				state, err := getSubvolumeState(ctx, clientsets, operatorNamespace, clusterNamespace, fs.Name, sv.Name, svg.Name)
+				// subvolume info returns error in case of pending clone or if it is not ready
+				// it is suggested to delete the pvc before deleting the subvolume.
+				if err != nil && errors.Is(err, syscall.EAGAIN) {
+					logging.Info("Subvolume info not found for: %q", err)
+					logging.Info("Please delete the pending pv if any before deleting the subvolume %s", sv.Name)
+					continue
+				}
```

Comment:
what if the error is not nil and not EAGAIN?

can you please test this and see if we are getting `EAGAIN` here to make sure we are good 👍🏻

Reply:
```
yatipadia:kubectl-rook-ceph$ ./bin/kubectl-rook-ceph subvolume ls
Filesystem  Subvolume  SubvolumeGroup  State
myfs csi-vol-9e760985-c5cd-46d1-8484-8ecfa1e3d6bc csi in-use
myfs csi-vol-5d13c38b-2263-46bc-826b-6c0df2ad4bcb csi in-use
Error: Error EAGAIN: subvolume 'csi-vol-71bb4f78-1373-4b7e-8400-ea5fd96a22dc' is not ready for operation info
. failed to run command. command terminated with exit code 11%!(EXTRA string=failed to get subvolume info)
Info: Subvolume info not found for: "Error EAGAIN: subvolume 'csi-vol-71bb4f78-1373-4b7e-8400-ea5fd96a22dc' is not ready for operation info\n. failed to run command. command terminated with exit code 11"
Info: Please delete the pending pv if any before deleting the 
```

Here's the output if the clone is in pending state , the info command is not ready incase the subvolume is not properly created, so this should be fine.

Reply:
@yati1998 in case of err !=nil and error is not syscall.EAGAIN we need to log it as error and let the user run the tool again, we should not ignore errors unless we know why we are doing like that

Reply:
done

Reply:
please review it again, thank you @Madhu-1